crates/host/src/lib.rs

```diff
@@ -63,34 +67,59 @@ impl<'a> TryFrom<&'a str> for ResourceRef<'a> {
                         .strip_prefix("bindle://")
                         .map(Self::Bindle)
                         .context("invalid Bindle reference"),
+                    "oci" => s
+                        .strip_prefix("oci://")
+                        .map(Self::Oci)
+                        .context("invalid OCI reference"),
                     // TODO: Support other schemes
                     scheme => bail!("unsupported scheme `{scheme}`"),
                 }
             }
-            Err(url::ParseError::RelativeUrlWithoutBase) => Ok(Self::Oci(s)), // TODO: Validate
+            Err(url::ParseError::RelativeUrlWithoutBase) => {
+                match Url::parse(&format!("oci://{s}")) {
+                    Ok(_url) => Ok(Self::Oci(s)),
+                    Err(e) => Err(anyhow!(e).context("failed to parse reference as OCI reference")),
+                }
+            }
             Err(e) => {
                 bail!(anyhow!(e).context(format!("failed to parse reference `{s}`")))
             }
         }
     }
 }
 
+fn extract_server<'a>(resource_ref: &ResourceRef<'a>) -> Option<&'a str> {
+    match resource_ref {
+        ResourceRef::File(_) => None,
+        ResourceRef::Bindle(resource_ref) | ResourceRef::Oci(resource_ref) => {
+            resource_ref.split('/').next()
+        }
+    }
+}
+
 /// Fetch an actor from a reference.
 #[instrument(skip(actor_ref))]
 pub async fn fetch_actor(
     actor_ref: impl AsRef<str>,
-    oci_opts: &oci::Config,
     allow_file_load: bool,
+    all_registry_settings: &HashMap<String, RegistrySettings>,
 ) -> anyhow::Result<Vec<u8>> {
-    match ResourceRef::try_from(actor_ref.as_ref())? {
+    let actor_ref = ResourceRef::try_from(actor_ref.as_ref())?;
+    let server = extract_server(&actor_ref).unwrap_or_default();
```

Comment:
I would expect to see something like 
```suggestion
    let server = actor_ref.authority().unwrap_or_default();
```